src/compression.rs

```diff
@@ -0,0 +1,226 @@
+//! Compression-related methods, all used in `axoasset::Local`
+
+use crate::error::*;
+use camino::Utf8Path;
+use flate2::{write::ZlibEncoder, Compression, GzBuilder};
+use std::{
+    fs::{self, DirEntry},
+    io::BufReader,
+};
+use xz2::write::XzEncoder;
+use zip::ZipWriter;
+
+/// Internal tar-file compression algorithms
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(crate) enum CompressionImpl {
+    /// .gz
+    Gzip,
+    /// .xz
+    Xzip,
+    /// .zstd
+    Zstd,
+}
+
+pub(crate) fn tar_dir(
+    src_path: &Utf8Path,
+    dest_path: &Utf8Path,
+    compression: &CompressionImpl,
+) -> Result<()> {
+    // Set up the archive/compression
+    // The contents of the zip (e.g. a tar)
+    let dir_name = src_path.file_name().unwrap();
+    let zip_contents_name = format!("{dir_name}.tar");
+    let final_zip_file = match fs::File::create(dest_path) {
```

Comment:
is this a spot where we could use `LocalAsset::write_new`?

Reply:
not sure, since we're just creating the file and not initially writing anything to it, before passing it off to the various buffer writers

Reply:
i think we could- and it would help with debugging as the error that throws is the write new error and so using the associated function i think makes sense

Reply:
hmmm, if anything, i'd rather add a new `write_and_open` method or something, since `write_new` doesn't actually return a `File` (which is what we need here). either that or we use `write_new` and then do `File::open` afterwards

Reply:
ahhh got it, sorry, haven't had any coffee yet today lol. i like the write and open idea but we could file that as an issue and not block on it